kolibri/plugins/epub_viewer/assets/src/views/EpubRendererIndex.vue

```diff
@@ -239,6 +237,12 @@
         }
         return null;
       },
+      visitedPages() {
+        if (this.extraFields && this.extraFields.contentState) {
+          return this.extraFields.contentState.visitedPages || [];
+        }
+        return [];
```

Comment:
May be a premature optimization, but we could implement this as an object, rather than an array.

This would mean when we set a new page on the visitedPages we'd just do:

`visitedPages[page] = true`

This means we don't need to do an O(n) lookup on an Array every time we check to see if the page is already in vistedPages.